miso-web/src/main/java/uk/ac/bbsrc/tgac/miso/webapp/controller/EditLibraryController.java

```diff
@@ -369,6 +376,55 @@ public String emPCRUnits() {
   public String emPCRDilutionUnits() {
     return emPCRDilution.UNITS;
   }
+  
+  @ModelAttribute("prepKits")
+  public List<KitDescriptor> getPrepKits() throws IOException {
+    List<KitDescriptor> list = new ArrayList<>(requestManager.listKitDescriptorsByType(KitType.LIBRARY));
+    Collections.sort(list, new Comparator<KitDescriptor>() {
+      @Override
+      public int compare(KitDescriptor kd1, KitDescriptor kd2) {
+        return kd1.getName().compareTo(kd2.getName());
+      }
+    });
+    return list;
+  }
+  
+  /**
+   * Adds child entities to a new detailed Library so they can be bound in the JSP. These will not all be useful for the same
+   * object, but are all included to accommodate the JSP.
+   * @param library
+   */
+  private void addNewDetailedLibraryEntities(Library library) {
+    library.setLibraryAdditionalInfo(new LibraryAdditionalInfoImpl());
+    library.getLibraryAdditionalInfo().setPrepKit(new KitDescriptor());
+  }
+
+  private void populateDesigns(ModelMap model, SampleClass sampleClass) throws IOException {
```

Comment:
We are getting to do this pattern uncomfortable often. I'm going to refactor this after.